### PR TITLE
[MT-5050] Suggestion - Tables

### PR DESCRIPTION
**Pull request by e1himself**
- Make naming consistent with other plugins and extensions
  - `TableEditor` &rarr; `TablesEditor`
  - `withTableEditor` &rarr; `withTables()`
  - `TableExtension` &rarr; `TablesExtension`
  - `TableRow` component &rarr; `TableRowElement`
- Drop unnecessary `focusEditor` from `slate-tables` schema. Instead rely on `ReactEditor` interface
- Refactor `slate-tables` extension to not depend on `type` property expectation
  - Introduce `is{Type}Node` and `create{Type}Node` functions to the schema
  - Drop `tableNodeTypes` schema prop
- Simplify header cell detection logic
- Add support for `first_column` header mode (to match `slate-tables` types expectations)
- Drop unnecessary dependencies from `slate-tables` package
- Implement Table nodes attributes normalization (to avoid noop differences in the document structure)

### CODE REVIEW

File: packages/slate-editor/src/extensions/table/components/elements/TableCellElement.tsx [hunk 1]
```diff
@@ -1,45 +0,0 @@
...
-                (cell) => cell === element,
-            );
-
-            if (isCurrentCellInFirstRow) {
-                return isCurrentCellInFirstRow;
-            }
-        }
-
-        return false;
-    }, [ctx.table, element]);
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/table/components/elements/TableCellElement.tsx:29]: This has been simplified to this: https://github.com/prezly/slate/pull/224/files#diff-052191ca2a1769b9020c25d8bed3e2a138d7e28ed977e5fba70f0b748f91fbe0R20-R25

File: packages/slate-editor/src/extensions/tables/components/elements/elements.module.scss [hunk 2]
```diff
@@ -28,4 +28,5 @@
     min-width: 40px;
     min-height: 40px;
     padding: $spacing-1-5;
+    outline: none !important;
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/tables/components/elements/elements.module.scss:31]: This fixes outline blinking when typing inside a cell.

File: packages/slate-editor/src/extensions/tables/components/elements/index.ts [hunk 3]
```diff
@@ -1,3 +1,3 @@
 export * from './TableCellElement';
 export * from './TableElement';
-export * from './TableRow';
+export * from './TableRowElement';
```
Comment (e1himself) [inline packages/slate-editor/src/extensions/tables/components/elements/index.ts:3]: The name was not consistent.

File: packages/slate-editor/src/modules/editor/lib/table/insertTable.ts [hunk 4]
```diff
@@ -10,7 +10,12 @@ export function insertTable(editor: Editor) {
         return;
     }
 
-    TableEditor.insertTable(editor, undefined, 3, 3);
+    TablesEditor.insertTable(editor, undefined, {
+        header: ['first_row'],
+        border: true,
+        rowsCount: 3,
+        columnsCount: 3,
+    });
```
Comment (e1himself) [inline packages/slate-editor/src/modules/editor/lib/table/insertTable.ts:18]: I've lifted the default table properties to the very top, to never apply them to a table created with default undefined properties.

File: packages/slate-tables/package.json [hunk 5]
```diff
@@ -52,8 +52,5 @@
         "react": "^16.9.0 || ^17.0.0 || ^18.0.0",
         "slate": "^0.81.0",
         "slate-react": "^0.81.0"
-    },
-    "dependencies": {
-        "@prezly/slate-commons": "^0.43.1"
```
Comment (e1himself) [inline packages/slate-tables/package.json:57]: It does not depend on `@prezly/slate-commons`

File: packages/slate-tables/src/TableEditor.tsx [hunk 6]
```diff
@@ -1,50 +0,0 @@
...
-import * as TableQueries from './queries';
-
-export interface TableNodeType {
-    table: string;
-    row: string;
-    cell: string;
-}
-
-export interface TableSchema {
-    tableNodeTypes: TableNodeType;
```
Comment (e1himself) [inline packages/slate-tables/src/TableEditor.tsx:14]: Replaced this with a set of `is{Type}Node` and `create{Type}Node` methods -- to not require the consumer app model having `type` property on nodes (same principle as with `slate-lists`).

File: packages/slate-tables/tsconfig.json [hunk 7]
```diff
@@ -8,8 +8,4 @@
     },
     "include": ["./src"],
     "exclude": ["node_modules"],
-    "references": [
-        { "path": "../slate-types/tsconfig.json" },
-        { "path": "../slate-commons/tsconfig.build.json" }
-    ]
```
Comment (e1himself) [inline packages/slate-tables/tsconfig.json:14]: These were not necessary.

File: packages/slate-types/src/nodes/TableNode.ts [hunk 8]
```diff
@@ -1,9 +1,17 @@
...
+export const TABLE_CELL_NODE_TYPE = 'table-cell';
+
+export type TableHeader = 'first_row' | 'first_column';
+
+export interface TableNode extends ElementNode {
+    type: typeof TABLE_NODE_TYPE;
+    children: TableRowNode[];
+    border?: boolean;
+    header?: TableHeader[];
+}
```
Comment (e1himself) [inline packages/slate-types/src/nodes/TableNode.ts:14]: Reordered them in logical order.

**Review by aspirisen — APPROVED** [review]

**Pull request merged**